src/server/survey_and_program_data.R

```diff
@@ -1,58 +1,83 @@
-library(tidyr)
-library(shiny)
-library(data.table)
-library(rhandsontable)
-library(first90)
+library(magrittr)
 
 getProgramDataInWideFormat <- function(country) {
-    data(prgm_dat)
     long <- prgm_dat[prgm_dat$country == country, ]
     long$country <- NULL
     long$notes <- NULL
-    wide <- spread(long, key = "type", value = "number")
+
+    if (nrow(long) == 0) {
+
+        # we create an empty data table here, but the plots require values, so we shouldn't let the user proceed
+        # unless they have at least one non-empty row
+        years <- seq(2005, 2017)
+        NbTested <- as.numeric(rep(NA, 2018-2005))
+        NbTestPos <- as.numeric(rep(NA, 2018-2005))
+        NbANCTested <- as.numeric(rep(NA, 2018-2005))
+        NBTestedANCPos <- as.numeric(rep(NA, 2018-2005))
+
+        wide <- data.frame(years, NbTested, NbTestPos, NbANCTested, NBTestedANCPos)
+        colnames(wide) <- c("year", "NbTested", "NbTestPos", "NbANCTested", "NBTestedANCPos")
+    }
+    else {
+        wide <- tidyr::spread(long, key = "type", value = "number")
+    }
+
     wide[c("year", "NbTested", "NbTestPos", "NbANCTested", "NBTestedANCPos")]
 }
 
 surveyAndProgramData <- function(input, output, state, spectrumFilesState) {
-    data(survey_hts)
-    data(prgm_dat)
+    data("survey_hts", package="first90")
+    data("prgm_dat", package="first90")
+
+    spectrumFilesState$newCountry <- FALSE
+
+    shiny::observeEvent(spectrumFilesState$newCountry, {
 
-    observeEvent(spectrumFilesState$country, {
```

Comment:
This was firing after the digest was read in, so over-writing the loaded survey and program data. Now listening for changes to `spectrumFileState$newCountry` instead, which only gets written to on upload of the first pjnz file.